text/1001-edgedb-install.rst

```diff
@@ -0,0 +1,298 @@
+
+``--version=<ver>``
+  Optionally specifies the server version to use.  If not specified,
+  the latest installed server version is used.
+
+``--start-conf=auto|manual``
+  If set to ``auto`` (the default), the server will be started automatically
+  on system boot.
+
+``--port=<port-number>``
+  Optionally specifies the port number on which the server should listen.
+
+``--system``
+  By default, ``edgedb server start`` runs the server in the user scope,
+  if ``--system`` is specified, it is started as a system-wide service
+  instead.
+
+``--server-options -- <options>``
+  Passes ``edgedb-server`` options verbatim.  Must be the last argument.
+
+
+edgedb server status
+====================
+
+Shows the status of the specified server instance or all instances.
+
+Synopsis
+--------
+
+``edgedb server status [options] [<name>]``
+
+Options
+-------
+
+``<name>``
+  The name of the EdgeDB instance.  If not specified, the name
+  ``default`` is used.
+
+``--all``
+  Show the status of all known instances.
+
+Implementation
+--------------
+
+The command outputs the state of the server instance
+(``running`` or ``stopped``), the port number it is configured to run on,
+the scope of the instance (system-wide or user-local), and the runtime under
+which the server is running (docker or native).
+
+
+edgedb server stop
+==================
+
+Stops the specified EdgeDB server instance.
+
+Synopsis
+--------
+
+``edgedb server stop [options] [<name>]``
+
+Options
+-------
+
+``<name>``
+  The name of the EdgeDB instance.  If not specified, the name
+  ``default`` is used.
+
+
+edgedb server upgrade
+=====================
+
+Upgrades the specified EdgeDB server instance to a given EdgeDB version.
+
+Synopsis
+--------
+
+``edgedb server upgrade [options] [<name>]``
+
+Options
+-------
+
+``<name>``
+  The name of the EdgeDB instance.  If not specified, the name
+  ``default`` is used.
+
+``--version``
+  Specifies the version of EdgeDB to upgrade to.  If not specified,
+  the latest available installed version is used.
+
+``--nightly``
+  Upgrade to a nightly release.
+
+``--allow-downgrade``
+  Allow downgrading to an older version.  Downgrades are prohibited by
+  default.
+
+``--revert``
+  Revert the upgrade if the original data directory has not been removed.
+
+Implementation
+--------------
+
+This command:
+
+* starts a temporary instance of the new EdgeDB server
+* pipes data from the old server with `dump`/`restore`
+* stops both servers, renames the data directories and restarts the new server.
+
+This keeps the original data directory in case ``--revert`` is requested.
+
+
+edgedb server prune
+===================
+
+Removes upgrade backups.
+
+Synopsis
+--------
+
+``edgedb server prune [options]``
+
+Options
+-------
+
+``--upgrade-backups``
+  Prune upgrade backups.  After this ``edgedb server upgrade --revert``
+  will be impossible.
+
+
+.. _design-considerations:
+
+Design Considerations
+=====================
+
+Instance names
```

Comment:
I'd pull this section to the top. Most of the commands descriptions refer to instances and everything was confusing up until I started reading this section.